### PR TITLE
MOB-3431 [Quote Summary] As a user, I can check the ETA, Price, vehic…

### PR DESCRIPTION
**Pull request by mateid-kr**
…le tags of my ride

## JIRA
[MOB-3431](https://karhoo.atlassian.net/browse/MOB-3431)

## DESCRIPTION
MOB-3431 [Quote Summary] As a user, I can check the ETA, Price, vehicle tags of my ride

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [X] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [X] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

File: uisdk/src/main/java/com/karhoo/uisdk/screen/booking/checkout/quotes/BookingQuotesPresenter.kt [hunk 1]
```diff
@@ -20,4 +27,45 @@ class BookingQuotesPresenter(val view: BookingQuotesMVP.View) : BookingQuotesMVP
...
+            if (shortVersion && index == 1) {
+                tagsText.append(" + ").append((tags.size - 1 - index).toString())
+                return tagsText
+            }
+        }
+
+        return tagsText
+    }
+
+    private fun getTagIcon(tag: String): Drawable? {
```
Comment (mohadian) [inline uisdk/src/main/java/com/karhoo/uisdk/screen/booking/checkout/quotes/BookingQuotesPresenter.kt:56]: This sounds like something we will be using in future. I would suggest creating an enum like `CardType` and then we can have an extension for it to return the icon or text

File: uisdk/src/main/java/com/karhoo/uisdk/screen/booking/checkout/quotes/BookingQuotesPresenter.kt [hunk 2]
```diff
@@ -20,4 +27,45 @@ class BookingQuotesPresenter(val view: BookingQuotesMVP.View) : BookingQuotesMVP
...
+        return when (tag.toLowerCase(Locale.ROOT)) {
+            "executive" -> view.getDrawableResource(R.drawable.kh_uisdk_ic_executive)
```
Comment (mohadian) [inline uisdk/src/main/java/com/karhoo/uisdk/screen/booking/checkout/quotes/BookingQuotesPresenter.kt:58]: Would it be better if we explicitly prefix them with `tag` them `kh_uisdk_ic_tag_`

File: uisdk/src/main/res/drawable/kh_uisdk_ic_electric.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    android:width="16dp"
+    android:height="16dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24">
+  <path
+      android:pathData="M13,2l-10,12l9,0l-1,8l10,-12l-9,0l1,-8z"
+      android:strokeLineJoin="round"
+      android:strokeWidth="2"
+      android:fillColor="#00000000"
+      android:strokeColor="@color/black"
```
Comment (mohadian) [inline uisdk/src/main/res/drawable/kh_uisdk_ic_electric.xml:11]: perhaps export this color and call it something like `kh_uisdk_tags_icon` or `kh_uisdk_icon` ? (same in other icon assets )

File: uisdk/src/main/java/com/karhoo/uisdk/screen/booking/checkout/quotes/BookingQuotesPresenter.kt (hunk 1 above)
Comment (mateid-kr) [reply uisdk/src/main/java/com/karhoo/uisdk/screen/booking/checkout/quotes/BookingQuotesPresenter.kt:56]: Added TagType object and moved the tag icon inside the class

File: uisdk/src/main/java/com/karhoo/uisdk/screen/booking/checkout/quotes/BookingQuotesPresenter.kt (hunk 2 above)
Comment (mateid-kr) [reply uisdk/src/main/java/com/karhoo/uisdk/screen/booking/checkout/quotes/BookingQuotesPresenter.kt:58]: Done

File: uisdk/src/main/res/drawable/kh_uisdk_ic_electric.xml (hunk 3 above)
Comment (mateid-kr) [reply uisdk/src/main/res/drawable/kh_uisdk_ic_electric.xml:11]: Done

Comment (mateid-kr) [reply uisdk/src/main/res/drawable/kh_uisdk_ic_electric.xml:11]: <color name="kh_uisdk_tag_icon_fill_color">#00000000</color>
 <color name="kh_uisdk_tag_icon_stroke_color">#000</color>

**Review by mohadian — APPROVED** [review]

**Pull request merged**